### ISSUE TITLE
IEDriverServer: Did not find any dialogs after dialog timeout

### ISSUE THREAD

**Issue by LightBit** [SeleniumHQ/selenium#6140]
## Meta -
OS: Windows 10 & Windows 7
Selenium Version: 3.12
Browser: Internet Explorer 11
IEDriverServer: 3.12.0.4, 3.13.0.2 (32-bit)

I randomly get timeout when trying to upload file with IE. Screenshot is created shortly after timeout where dialog can be seen. I have tried to increase file upload dialog timeout, but it did not help (it did wait longer).

Is there any idea what to do, to get more info or workaround?
Is it possible that sometimes dialog is opened from another IE process than expected or something like that?

## Expected Behavior -
Selected file to upload.

## Actual Behavior -
Random timeout:
sendkeyscommandhandler.cpp(169) Clicking upload button and starting to handle file selection dialog
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(312) Found no dialogs as children of parent window
sendkeyscommandhandler.cpp(572) Did not find any dialogs after dialog timeout

## Steps to reproduce -
Upload file many times?

**Comment by LightBit** [comment]
Just tried with IEDriverServer 3.13.0.2 (because there were some changes on upload) and it has the same problem.

**Comment by jimevans** [comment]
@LightBit File uploads are not globally broken for IE. Without a full, reliable reproduction case, it will be nearly impossible to track down where the issue might be. It’s likely that the dialog identification is flawed in attempting to locate the file selection dialog, but that may be browser-version and even OS-version specific.

**Comment by LightBit** [comment]
If anyone has the same problem, here is workaround in Python:
```
try:
    element.send_keys(file_name)
except socket.timeout:
    import uiautomation
    uiautomation.TIME_OUT_SECOND = 5
    dialog = uiautomation.WindowControl(Name='Choose File to Upload')
    dialog.EditControl(Name='File name:').SetValue(file_name)
    dialog.ButtonControl(Name='Open', AutomationId='1').Click()
```
It is limited to English and assumes only one upload dialog at a time, but it works for me.

**Comment by dank8** [comment]
It appears that some "Choose File to Upload" use a SplitButtonControl rather than a ButtonControl i have been unable to get the switch_to_alert().accept() , to work, the file window has a dropdown on the field name text edit, which has two buttons with name "Open" (one with the File combo box and the other with the Open SplitButtonControl. 

IE Version 11.2580.14393.0
Windows Server 2016 Standard

@LightBit thanks for your example, the first uiautomation with python3 i could find, updated to show file dialog working example:

     import uiautomation as automation

     print(automation.GetRootControl())
     uiWindow = automation.WindowControl(searchDepth = 2, Name = 'Choose File to Upload')
     print(uiWindow.Name)
     uiWindow.SetTopmost(True)
     fileName = uiWindow.EditControl(searchDepth = 4,Name = 'File name:')
     fileName.SendKeys("State of testing.MicroFocus.pdf")
     openFile = uiWindow.SplitButtonControl(searchDepth = 3,Name = 'Open')
     openFile.Click()

**Comment by LightBit** [comment]
Duplicate of #6991.
It seems to be fixed with 400c32a457b883dbb20066314788b97508547b99.
